skrf/circuit.py

```diff
@@ -1562,6 +1673,27 @@ def invalide_to_reduce(cnx: list[tuple[Network, int]]) -> bool:
 
         connections = tmp_cnxs
 
+    if split_multi:
+        tmp_cnxs = []
+
+        for cnx in connections:
+            # Check if the connection has more than 2 components
+            if len(cnx) <= 2:
+                tmp_cnxs.append(cnx)
+                continue
+
+            # Create a splitter for the connection
+            _media = media.DefinedGammaZ0(cnx[0][0].frequency)
+            splitter = _media.splitter(
+                name=f"Splt_{'&'.join(ntwk.name for ntwk, _ in cnx)}", nports=len(cnx)
+            )
```

Comment:
what about non-50ohm char impedance cases?

Reply:
In non-50 Ohm characteristic impedance scenarios, a **mismatched impedance** `splitter` replaces the **uniform impedance** `splitter` to eliminate the reflections between the `splitter` and corresponding `Network`.

Even with a **uniform impedance** `splitter`, equal outcomes can be achieved as it shifts internal reflections to the interface between the `splitter` and the `Network`.

Here are the examples:

```python3
import skrf as rf
import numpy as np


def gen_s_params(p_n: int = 2, f_n: int = 101):
    return np.random.randn(f_n, p_n, p_n) + 1j * np.random.randn(f_n, p_n, p_n)


frequency_number = 101
freq = rf.Frequency(1, 100, frequency_number, "ghz")


ntwkA = rf.Network(
    frequency=freq,
    z0=40,
    s=gen_s_params(2, frequency_number),
    name="ntwkA",
)
ntwkB = rf.Network(
    frequency=freq,
    z0=50,
    s=gen_s_params(2, frequency_number),
    name="ntwkB",
)
ntwkC = rf.Network(
    frequency=freq,
    z0=60,
    s=gen_s_params(2, frequency_number),
    name="ntwkC",
)

port1 = rf.Circuit.Port(frequency=freq, name="port1")
port2 = rf.Circuit.Port(frequency=freq, name="port2")
port3 = rf.Circuit.Port(frequency=freq, name="port3")

# Directly build circuit
dir_cnx = [
    [(port1, 0), (ntwkA, 0)],
    [(port2, 0), (ntwkB, 0)],
    [(port3, 0), (ntwkC, 0)],
    [(ntwkA, 1), (ntwkB, 1), (ntwkC, 1)],
]

# Mismatch splitter circuit
line = rf.media.DefinedGammaZ0(frequency=freq)
mis_splitter = line.splitter(nports=3, name="mis_splitter", z0=[40, 50, 60])
mis_cnx = [
    [(port1, 0), (ntwkA, 0)],
    [(port2, 0), (ntwkB, 0)],
    [(port3, 0), (ntwkC, 0)],
    [(ntwkA, 1), (mis_splitter, 0)],
    [(ntwkB, 1), (mis_splitter, 1)],
    [(ntwkC, 1), (mis_splitter, 2)],
]

# Uniform splitter circuit
line = rf.media.DefinedGammaZ0(frequency=freq)
uni_splitter = line.splitter(nports=3, name="mis_splitter", z0=50)
uni_cnx = [
    [(port1, 0), (ntwkA, 0)],
    [(port2, 0), (ntwkB, 0)],
    [(port3, 0), (ntwkC, 0)],
    [(ntwkA, 1), (uni_splitter, 0)],
    [(ntwkB, 1), (uni_splitter, 1)],
    [(ntwkC, 1), (uni_splitter, 2)],
]


# Create circuit object
dir_ckt = rf.Circuit(dir_cnx)
mis_ckt = rf.Circuit(mis_cnx)
uni_ckt = rf.Circuit(uni_cnx)

# Differential network
print(np.allclose(dir_ckt.s_external, mis_ckt.s_external))
print(np.allclose(dir_ckt.s_external, uni_ckt.s_external))
```
And the output:
```shell
True
True
```